docs: restructures installation and moves legacy setups (#2111)
shopware/docs: guides/installation/docker-setup.md

  - Ensure Docker has permission to bind to local ports (typically:80 or:8080) and is not already bound to conflicting ports
- `make` is installed on your machine (`apt install make` on Ubuntu, `brew install make` on macOS)
- `Docker Compose` is installed on your machine. Docker Desktop provides it automatically. If you're using OrbStack or something else, you can follow the official [Docker Compose installation guide](https://docs.docker.com/compose/install/).
- Enough disk and network capacity to pull images (~500MB+ per image depending on tags)
- [Git](https://git-scm.com/) installed and available in your `PATH`

This setup comes preinstalled with the [Shopware CLI](../../../products/cli/index.md), which helps with building, refactoring, validating, and managing Shopware projects and extensions. It works with all setups and is used in most Shopware upgrade, build, and CI workflows. The CLI is available in the container shell.

## Pre-pull the image (optional)

If you haven’t yet downloaded the Shopware Docker image, pull it now:

```bash
docker pull ghcr.io/shopware/docker-dev:php8.3-node24-caddy
```

> If you skip this step, Docker will automatically download the image during project creation. That’s normal, but pre-pulling makes the process cleaner and enables you to avoid waiting for large image downloads.

Reviewer: ```suggestion
If this step is skipped, Docker automatically downloads the image during project creation. Preloading can make the process cleaner and help avoid waiting for large image downloads.
```
